test/boost/secondary_index_test.cc

```diff
@@ -1208,6 +1208,279 @@ SEASTAR_TEST_CASE(test_indexing_paging_and_aggregation) {
+    eventually([&] { 
+        auto res = e.execute_cql("SELECT count(*) " + rest_of_query).get0();
+        assert_that(res).is_rows().with_rows({
+            { long_type->decompose(expected_count) }
+        });
+        res = e.execute_cql("SELECT * " + rest_of_query).get0();
+        assert_that(res).is_rows().with_size(expected_count);
+    });
+}
+
+static constexpr int page_scenarios_row_count = 2 * cql3::statements::select_statement::DEFAULT_COUNT_PAGE_SIZE + 120;
+static constexpr int page_scenarios_initial_count = 3;
+static constexpr int page_scenarios_window_size = 4;
+static constexpr int page_scenarios_just_before_first_page = cql3::statements::select_statement::DEFAULT_COUNT_PAGE_SIZE - page_scenarios_window_size;
+static constexpr int page_scenarios_just_after_first_page = cql3::statements::select_statement::DEFAULT_COUNT_PAGE_SIZE + page_scenarios_window_size;    
+static constexpr int page_scenarios_just_before_second_page = 2 * cql3::statements::select_statement::DEFAULT_COUNT_PAGE_SIZE - page_scenarios_window_size;
+static constexpr int page_scenarios_just_after_second_page = 2 * cql3::statements::select_statement::DEFAULT_COUNT_PAGE_SIZE + page_scenarios_window_size;    
+
+static_assert(page_scenarios_initial_count < page_scenarios_row_count);
+static_assert(page_scenarios_window_size < cql3::statements::select_statement::DEFAULT_COUNT_PAGE_SIZE);
+static_assert(page_scenarios_just_after_second_page < page_scenarios_row_count);
+
+// Executes `insert` lambda page_scenarios_row_count times. 
+// Runs `validate` lambda in a few scenarios:
+//
+// 1. After a small number of `insert`s
+// 2. In a window from just before and just after `insert`s where executed
+//    DEFAULT_COUNT_PAGE_SIZE times
+// 3. In a window from just before and just after `insert`s where executed
+//    2 * DEFAULT_COUNT_PAGE_SIZE times
+// 4. After all `insert`s
+void test_with_different_page_scenarios(
+    noncopyable_function<void (int64_t)> insert, noncopyable_function<void (int64_t)> validate) {
+
+    int64_t current_row = 0;
+    for (; current_row < page_scenarios_initial_count; current_row++) {
+        insert(current_row);
+        validate(current_row + 1);
+    }
+
+    for (; current_row < page_scenarios_just_before_first_page; current_row++) {
+        insert(current_row);
+    }
+
+    for (; current_row < page_scenarios_just_after_first_page; current_row++) {
+        insert(current_row);
+        validate(current_row + 1);
+    }
+
+    for (; current_row < page_scenarios_just_before_second_page; current_row++) {
+        insert(current_row);
+    }
+
+    for (; current_row < page_scenarios_just_after_second_page; current_row++) {
+        insert(current_row);
+        validate(current_row + 1);
+    }   
+
+    for (; current_row < page_scenarios_row_count; current_row++) {
+        insert(current_row);
+    }
+
+    validate(current_row);
+}
+
+SEASTAR_TEST_CASE(test_secondary_index_on_ck_first_column_and_aggregation) {
```

Comment:
I don't like the tests in a separate commit.  It complicates bisection and backporting: `git bisect` will think that the code commit is OK because of no failures; a maintainer may backport the code but not the tests into a branch that will then lack coverage.

Reply:
Maintainers should not generally backport single commits belonging to a larger series, but rather the whole series, so I'm personally fine with this being in a separate commit. No strong opinion either way though.

Reply:
I like the separation. The smaller the commits the easier it is to review them.

Reply:
When reading some older @psarna commits, I noticed that he separated tests into a separate commit: https://github.com/scylladb/scylla/commit/3d9a37f28fe413bb310059eb4d8d401e032d24b1. I thought this was some convention that should be followed.

Reply:
Yup, I tend to do that, because splitting to more commits makes rebasing easier.

Reply:
I do that as well as I recommend doing it.

Reply:
OK, OK. :)